src/main/kotlin/org/rust/ide/utils/SearchByOffset.kt

```diff
@@ -62,8 +63,31 @@ fun findExpressionInRange(file: PsiFile, startOffset: Int, endOffset: Int): RsEx
  * as return expression, doesn't allow attrs and items) within selected range.
  */
 fun findStatementsInRange(file: PsiFile, startOffset: Int, endOffset: Int): Array<out PsiElement> {
-    var (element1, element2) = file.getElementRange(startOffset, endOffset) ?: return emptyArray()
+    val (element1, element2) = file.getElementRange(startOffset, endOffset) ?: return emptyArray()
 
+    return findStatementsInRange(element1, element2)
+}
+
+/**
+ * Finds statements (mostly [RsStmt]s, [PsiComment]s and [RsExpr]
+ * as return expression, doesn't allow attrs and items) or a single expr within selected range.
+ */
+fun findStatementsOrExprInRange(file: PsiFile, startOffset: Int, endOffset: Int): Array<out PsiElement> {
+    val (element1, element2) = file.getElementRange(startOffset, endOffset) ?: return emptyArray()
+    val parent = PsiTreeUtil.findCommonParent(element1, element2) ?: return emptyArray()
+
+    val mostDistantParent = parent.ancestors
```

Comment:
Maybe `val mostDistantParent = parent.ancestors.last { it.textRange == parent.textRange }`?
`lastOrNull` is redundant here because there is at least one such ancestor — the `parent` itself.